idaes_examples/notebooks/docs/unit_models/custom_unit_models/creating_unit_model.ipynb

```diff
@@ -0,0 +1,2296 @@
+    "        obj.add_default_units(\n",
+    "            {\n",
+    "                \"time\": units.hour,\n",
+    "                \"length\": units.m,\n",
+    "                \"mass\": units.g,\n",
+    "                \"amount\": units.mol,\n",
+    "                \"temperature\": units.K,\n",
+    "            }\n",
+    "        )"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "4c942c2b",
+   "metadata": {},
+   "source": [
+    "# 1.3 State Block\n",
+    "\n",
+    "After the `PhysicalParameterBlock` class has been created, the next step is to write the code necessary to create the State Blocks that will be used throughout the flowsheet. `StateBlock` contains all the information necessary to define the state of the system. This includes the state variables and constraints on those variables which are used to describe a state property like the enthalpy, material balance, etc.\n",
+    "\n",
+    "Creating a State Block requires us to write two classes. The reason we write two classes is because of the inherent nature of how `declare_process_block_data` works. `declare_process_block_data` facilitates creating an `IndexedComponent` object which can handle multiple `ComponentData` objects which represent the component at each point in the indexing set. This makes it easier to build an instance of the model at each indexed point. However, State Blocks are slightly different, as they are always indexed (at least by time). Due to this, we often want to perform actions on all the elements of the indexed StateBlock all at once (rather than element by element).\n",
+    "\n",
+    "The class `_StateBlock` is defined without the `declare_process_block_data` decorator and thus works as a traditional class and this facilitates performing a method on the class as a whole rather than individual elements of the indexed property blocks. In this class we define the `release_state` function. `release_states` function is to used to free the state variable from the current state. It takes flags as the argument which determine if the state is to held or can be released. This function uses `revert_state_vars` function which reverts the fixed state of the state variables within an IDAES StateBlock based on a set of flags of the previous state. After releasing the state variables, completion of state release is logged.\n",
+    "\n",
+    "The above function comprise of the _StateBlock, next we shall see the construction of the OrgPhaseStateBlockData class."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "id": "da8106f0",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "class _StateBlock(StateBlock):\n",
+    "    \"\"\"\n",
+    "    This Class contains methods which should be applied to Property Blocks as a\n",
+    "    whole, rather than individual elements of indexed Property Blocks.\n",
+    "    \"\"\"\n",
+    "    def release_state(self, flags, outlvl=idaeslog.NOTSET):\n",
```

Comment:
You should be able to remove all of this too as this is the old initialization API (which we do not want to advertise any more).